core/common/src/main/java/alluxio/master/file/meta/PersistenceState.java

```diff
@@ -22,4 +25,59 @@
   TO_BE_PERSISTED, // the file is to be persisted in the under FS
   PERSISTED, // the file is persisted in the under FS
   LOST // the file is lost but not persisted in the under FS
+  ;
+
+  /**
+   * Encode a state into a single byte.
+   *
+   * @return the state encoded as a byte
+   */
+  public byte encode() {
```

Comment:
Is all this new code for "replacing" the current persistence state? I don't think we are ready to replace that yet?

Reply:
So it does not need to be included in this PR, but I gave it as a sample of how we could encode the state in a small portion of memory